pkg/monitor/cluster/maintenance_test.go

```diff
@@ -0,0 +1,97 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/metrics"
+	mock_metrics "github.com/Azure/ARO-RP/pkg/util/mocks/metrics"
+)
+
+func TestEmitPucmState(t *testing.T) {
```

Comment:
Can we split these state tests into separate table-based test cases instead of doing them all in one case?

```suggestion
func TestEmitPucmState(t *testing.T) {
    tests := []struct {
        name string
        state api.ProvisioningState
        pucmOngoing bool
        wantGauge map[string]string
    } {
        name: "unplanned ongoing",
        state: api.ProvisioningStateAdminUpdating,
        wantGauge: map[string]string {"state": pucmUnplannedOngoing.String()},
    }
    etc...
```